app/presenters/tree_builder_ops_rbac_features.rb

```diff
@@ -0,0 +1,160 @@
+class TreeBuilderOpsRbacFeatures < TreeBuilder
+  has_kids_for Menu::Section,     [:x_get_tree_section_kids]
+  has_kids_for Menu::Item,        [:x_get_tree_item_kids]
+  has_kids_for MiqProductFeature, [:x_get_tree_feature_kids]
+
+  attr_reader :role, :features, :editable, :root_counter
+
+  def initialize(name, type, sandbox, build, role:, editable: false)
+    @role     = role
+    @editable = editable
+    @features = @role.miq_product_features.order(:identifier).pluck(:identifier)
+
+    @root_counter = []
+
+    # Make sure tree_state doesn't hold on to old data between requests
+    TreeState.new(sandbox).remove_tree(name)
+
+    super(name, type, sandbox, build)
+  end
+
+  private
+
+  def filtered_menus
+    menus = Menu::Manager.map do |section|
+      next if section.id == :cons && !Settings.product.consumption
+      next if section.name.nil?
+      next unless Vmdb::PermissionStores.instance.can?(section.id)
+
+      section
+    end
+
+    menus
+  end
+
+  def set_locals_for_render
+    locals = {
+      :checkboxes   => true,
+      :three_checks => true,
+      :check_url    => "/ops/rbac_role_field_changed/",
+      :onclick      => nil
+    }
+
+    if editable
+      locals[:oncheck] = "miqOnCheckHandler"
+    end
+
+    super.merge!(locals)
+  end
+
+  def x_get_tree_roots(count_only = false, _options)
+    top_nodes = filtered_menus
+    top_nodes << MiqProductFeature.obj_features["all_vm_rules"][:feature]
+
+    count_only_or_objects(count_only, top_nodes)
+  end
+
+  def x_get_tree_section_kids(parent, count_only = false)
+    kids = parent.items.reject do |item|
+      if item.kind_of?(Menu::Item)
+        item.feature.nil? || !MiqProductFeature.feature_exists?(item.feature)
+      end
+    end
+
+    count_only_or_objects(count_only, kids)
+  end
+
+  def x_get_tree_item_kids(parent, count_only = false)
+    kids = MiqProductFeature.obj_feature_children(parent.feature)
+    count_only_or_objects(count_only, kids)
+  end
+
+  def x_get_tree_feature_kids(parent, count_only = false)
+    kids = MiqProductFeature.obj_feature_children(parent.identifier) || []
+    count_only_or_objects(count_only, kids)
+  end
+
+  def tree_init_options(_tree_name)
+    {
+      :lazy           => false,
+      :add_root       => true,
+      :role           => role,
+      :features       => features,
+      :editable       => editable,
+      :node_id_prefix => node_id_prefix
+    }
+  end
+
+  def root_options
```

Comment:
Do we really need two methods (`root_node`, `root_options`) here? If yes, please use [aliasing](http://apidock.com/ruby/Module/alias_method).